examples/MicroserviceExample/WorkerService/RabbitMqConsumer.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Trace;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace WorkerService
+{
+    public class RabbitMqConsumer : BackgroundService
```

Comment:
It seems we have more code showing how to use `RabbitMq` than OpenTelemetry itself.
Wonder if there is something simpler, that helps people to focus on the custom propagation part.

Reply:
Yes, I had a similar thought. The applications required a fair amount of boilerplate to get RabbitMQ functionality in place. I can spend a bit of time refactoring things a bit to see if I can get the OpenTelemetry part of it all better highlighted.

Reply:
Agree that it takes some time to actually get to the OTel tracing/propagation part in the jungle of RabbitMq and boiler plate codes..

Reply:
Tough crowd, eh? Nice work @alanwest! Messaging is one of the more complex/advanced cases. Maybe we just need another sample doing something more trivial like calling a web service?

Reply:
> Tough crowd, eh?

😄 not at all, I appreciate the input. Working this afternoon to see what I can do to better highlight the OTel pieces. Stay tuned!

> maybe we just need another sample doing something more trivial like calling a web service?

You thinking like a call demonstrating HttpClient instrumentation or something?

Reply:
> You thinking like a call demonstrating HttpClient instrumentation or something?

That's what I was thinking. But we do kind of already have one in there:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f01bdc1080ec6a327eeffd76d8189edaef090b2/examples/AspNetCore/Controllers/WeatherForecastController.cs#L24-L27

Reply:
I've made an effort to separate the OpenTelemetry logic from the RabbitMQ 🌴 🐰  jungle 🐇 🌴.

Basically there are two classes now that are pretty succinct and I think put a pretty good focus on the context propagation:
* The WebApi has a `MessageSender` class
* The WorkService has a `MessageProcessor` class

Reply:
> > You thinking like a call demonstrating HttpClient instrumentation or something?
> 
> That's what I was thinking. But we do kind of already have one in there:

I'm definitely a fan of a more turnkey, holistic example that showcases a number of things. How about after we get some consensus on the direction of this sample, we can discuss what else may make sense to take on, and then do it in smaller follow up PRs?